ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh

```diff
@@ -30,7 +30,44 @@ esac
 
 dir=/tmp/installer
 mkdir "${dir}/"
-cp "${SHARED_DIR}/install-config.yaml" "${dir}/"
+
+function set_install_config_one_replica () {
+  pip3 install pyyaml --user
+  python3 -c '
+import yaml
+import sys
+
+with open(sys.argv[1]) as f:
+    cfg = yaml.safe_load(f)
+
+# Some workflows do not define controlPlane in install-config.yaml
+if not "controlPlane" in cfg:
+  cfg["controlPlane"] = {}
+
+cfg["controlPlane"]["replicas"] = 1
+
+# Some workflows do not define any compute machine pools in install-config.yaml
+if not "compute" in cfg:
+    cfg["compute"] = [
+        {
+            "name": "worker",
+            "platform": {},
+        }
+    ]
+
+for machine_pool in cfg["compute"]:
+    machine_pool["replicas"] = 0
+
+with open(sys.argv[2], "w") as f:
+    yaml.safe_dump(cfg, f)
+' "$@"
+}
+
+if [[ $CLUSTER_VARIANT == "single-node" ]]; then
+  set_install_config_one_replica "${SHARED_DIR}/install-config.yaml" "${dir}/install-config.yaml"
+else
+  cp "${SHARED_DIR}/install-config.yaml" "${dir}/"
+fi
```

Comment:
Maybe a simple mutation will make it more generic and more comfortable for extensions:
```suggestion
if [[ $CLUSTER_VARIANT == "single-node" ]]; then
  set_install_config_one_replica "${SHARED_DIR}/install-config.yaml"
fi
cp "${SHARED_DIR}/install-config.yaml" "${dir}/"
```